elastic-buffer-sim/src/Bittide/Simulate.hs

```diff
@@ -84,65 +85,156 @@ tunableClockGen settlePeriod periodOffset stepSize _reset speedChange =
     in
       newPeriod :- go newSettleCounter newPeriod scs
 
--- | Determines how 'elasticBuffer' should respond to underflow/overflow.
-data OverflowMode
-  -- | Saturate at empty/full boundaries. Useful for testing purposes.
-  = Saturate
-  -- | Error on write-when-full, or read-when-empty. This mode shuold be used in
-  -- practise. Overflowing elastic buffer cannot happen in a real Bittide system.
-  | Error
+type ResetClockControl = Bool -- request a reset in other elastic buffers
+type HasUnderflowed = Bool
+type HasOverflowed = Bool
+type DisableTilHalf = Bool
+type ForceReset = Bool -- force a "reset" to EB midpoint
+type PropagateReset = Bool
+type ResetEbs = Bool
 
--- | Simple model of a FIFO that only models the interesting part for conversion:
--- data counts.
+type DisableWrites = Bool
+type DisableReads = Bool
+
+data WrResetDomain = Wait -- ^ after reset/when nothing has overflowed
+                   | DisableWritesEnableReads
+                   | EnableWritesDisableReads
+                   deriving (Generic, NFDataX)
+
+-- | This wrapper disables reads or writes when the elastic buffer
+-- over-/underflows, as appropriate.
+--
+-- It also censors 'DataCount' after an over-/underflow, so that we do not take
+-- measurements from an elastic buffer until it has settled back to its
+-- midpoint.
+ebController ::
+  forall readDom writeDom.
+  (KnownDomain readDom, KnownDomain writeDom) =>
+  ElasticBufferSize ->
+  Clock readDom ->
+  Reset readDom ->
+  Enable readDom ->
+  Clock writeDom ->
+  Reset writeDom ->
+  Enable writeDom ->
+  (Signal readDom DataCount, Signal readDom ForceReset, Signal readDom ResetClockControl)
+ebController size clkRead rstRead enaRead clkWrite rstWrite enaWrite =
+  unbundle
+    (go <$> rdToggle <*> outRd <*> overflowRd <*> requestRst)
+ where
+  rstReadS = unsafeToHighPolarity rstRead
+
+  -- tie off reset to mealy while we do our "reset" dance
+  ne'erRst = unsafeFromHighPolarity (pure False)
```

Comment:
I've noted this in my larger review, but we weary whenever you write this. Good change you actually wanted to use a specialized function (e.g., `delay`) or not want this at all.